nodebuilder/settings.go

```diff
@@ -105,13 +110,58 @@ func WithMetrics(metricOpts []otlpmetrichttp.Option, nodeType node.Type, buildIn
 	return opts
 }
 
+func WithTraces(opts []otlptracehttp.Option, pyroOpts []otelpyroscope.Option) fx.Option {
+	options := fx.Options(
+		fx.Supply(opts),
+		fx.Supply(pyroOpts),
+		fx.Invoke(initializeTraces),
+	)
+	return options
+}
+
+func initializeTraces(
+	ctx context.Context,
+	nodeType node.Type,
+	peerID peer.ID,
+	network p2p.Network,
+	buildInfo *node.BuildInfo,
+	opts []otlptracehttp.Option,
+	pyroOpts []otelpyroscope.Option,
+) error {
+	var tp trace.TracerProvider
```

Comment:
nit but since you declare this anyway, just make a switch on `len(pyroOpts)` instead of setting tp the way you're doing now ? Also can you declare tp closer to where you set it?